platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/JsonBundle.java

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.application.service.migratable;
+
+import org.apache.commons.lang.Validate;
+import org.boon.json.annotations.JsonIgnore;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.Version;
+
+/** Defines a Json object to represent an exported bundle. */
+public class JsonBundle implements JsonValidatable {
+  public static String getFullName(Bundle bundle) {
+    return bundle.getSymbolicName() + '/' + bundle.getVersion();
+  }
+
+  public static SimpleState getSimpleState(Bundle bundle) {
+    return JsonBundle.getSimpleState(bundle.getState());
+  }
+
+  private static SimpleState getSimpleState(int state) {
+    if (state == Bundle.UNINSTALLED) {
+      return SimpleState.UNINSTALLED;
+    } else if ((state == Bundle.STARTING) || (state == Bundle.ACTIVE)) {
+      return SimpleState.ACTIVE;
+    } // else - INSTALLED OR STOPPING
+    return SimpleState.INSTALLED;
+  }
+
+  private final String name;
+
+  private final Version version;
+
+  private final Long id; // Long used to detect missing Json entries as null
```

Comment:
Yes. The other ones used Nullable when the attribute is optional and can be null. Here the attribute is not optional and shouldn't be null. Now the part I hate about Boon is that because Boon will not set attributes that are not define in the Json and bypass `final` or constructors, this attribute can be left null even though I do not let that happen in the existing constructor. Therefore the `validate()` is the only place it will check for null and throw an exception.
I will mark it `@Nullable` and a comment about Boon.